app/adaptive_tessellation/AdaptiveTessellation.cpp

```diff
@@ -869,77 +907,97 @@ double AdaptiveTessellation::get_area_accuracy_error_per_face_triangle_matrix(
+    error_after_1 = new_computed_errors[0];
+    error_after_2 = new_computed_errors[1];
+    if (edge_tuple.switch_face(*this).has_value()) {
+        const Eigen::Vector2f uv4 =
+            vertex_attrs
+                [edge_tuple.switch_face(*this).value().switch_edge(*this).switch_vertex(*this).vid(
+                     *this)]
+                    .pos.cast<float>();
+
+        new_triangles[0] = {uv1(0), uv1(1), mid_point_uv(0), mid_point_uv(1), uv4(0), uv4(1)};
+        new_triangles[1] = {uv2(0), uv2(1), mid_point_uv(0), mid_point_uv(1), uv4(0), uv4(1)};
         m_texture_integral.get_error_per_triangle(new_triangles, new_computed_errors);
-        error_after_1 = new_computed_errors[0];
-        error_after_2 = new_computed_errors[1];
-        if (edge_tuple.switch_face(*this).has_value()) {
-            const Eigen::Vector2f uv4 = vertex_attrs[edge_tuple.switch_face(*this)
-                                                         .value()
-                                                         .switch_edge(*this)
-                                                         .switch_vertex(*this)
-                                                         .vid(*this)]
-                                            .pos.cast<float>();
-
-            new_triangles[0] = {uv1(0), uv1(1), mid_point_uv(0), mid_point_uv(1), uv4(0), uv4(1)};
-            new_triangles[1] = {uv2(0), uv2(1), mid_point_uv(0), mid_point_uv(1), uv4(0), uv4(1)};
-            m_texture_integral.get_error_per_triangle(new_triangles, new_computed_errors);
-
-            error_after_3 = new_computed_errors[0];
-            error_after_4 = new_computed_errors[1];
-        } else {
-            // TODO set error after 3 and 4 to the the mirror face error when it's seam
-            error_after_3 = error_after_1;
-            error_after_4 = error_after_2;
-        }
-        e_after = error_after_1 + error_after_2 + error_after_3 + error_after_4;
-        error = e_before - e_after;
+
+        error_after_3 = new_computed_errors[0];
+        error_after_4 = new_computed_errors[1];
+    } else {
+        // TODO set error after 3 and 4 to the the mirror face error when it's seam
+        error_after_3 = error_after_1;
+        error_after_4 = error_after_2;
     }
+    e_after = error_after_1 + error_after_2 + error_after_3 + error_after_4;
+    error = e_before - e_after;
     return {error, error1, error2};
 }
 
+double AdaptiveTessellation::get_one_ring_quadrics_error_for_vertex(const Tuple& v) const
+{
+    double ret = 0.0;
+    wmtk::Quadric<double> q;
+    for (const Tuple& tri : get_one_ring_tris_for_vertex(v)) {
+        q += get_face_attrs(tri).accuracy_measure.quadric;
+    }
+    auto v_pos = vertex_attrs[v.vid(*this)].pos;
+    Eigen::Matrix<double, 3, 1> v_world_pos =
+        mesh_parameters.m_displacement->get(v_pos(0), v_pos(1));
+    ret = v_world_pos.transpose() * q.A() * v_world_pos - 2.0 * v_world_pos.dot(q.b()) + q.c;
```

Comment:
You can simply calculate `ret = q(v_world_pos);` instead (for smoothing I do explicitly because of autodiff).

Reply:
changing